packages/features/ee/workflows/components/AgentConfigurationSheet.tsx

```diff
@@ -0,0 +1,1026 @@
+import { Controller, useForm } from "react-hook-form";
+import { z } from "zod";
+
+import { Dialog } from "@calcom/features/components/controlled-dialog";
+import { formatPhoneNumber } from "@calcom/lib/formatPhoneNumber";
+import { useLocale } from "@calcom/lib/hooks/useLocale";
+import { PhoneNumberSubscriptionStatus } from "@calcom/prisma/enums";
+import type { RouterOutputs } from "@calcom/trpc/react";
+import { trpc } from "@calcom/trpc/react";
+import { Alert } from "@calcom/ui/components/alert";
+import { Badge } from "@calcom/ui/components/badge";
+import { Button } from "@calcom/ui/components/button";
+import { DialogContent, DialogHeader, DialogFooter as BaseDialogFooter } from "@calcom/ui/components/dialog";
+import { ConfirmationDialogContent } from "@calcom/ui/components/dialog";
+import { Dialog as UIDialog } from "@calcom/ui/components/dialog";
+import {
+  Dropdown,
+  DropdownItem,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  DropdownMenuTrigger,
+} from "@calcom/ui/components/dropdown";
+import { AddVariablesDropdown } from "@calcom/ui/components/editor";
+import { ToggleGroup, Switch } from "@calcom/ui/components/form";
+import { Label, TextArea, Input, TextField, Form } from "@calcom/ui/components/form";
+import { Icon } from "@calcom/ui/components/icon";
+import {
+  Sheet,
+  SheetContent,
+  SheetHeader,
+  SheetTitle,
+  SheetBody,
+  SheetFooter,
+} from "@calcom/ui/components/sheet";
+import { showToast } from "@calcom/ui/components/toast";
+import { Tooltip } from "@calcom/ui/components/tooltip";
+
+import { DYNAMIC_TEXT_VARIABLES } from "../lib/constants";
+import { TestAgentDialog } from "./TestAgentDialog";
+
+const agentSchema = z.object({
+  generalPrompt: z.string().min(1, "General prompt is required"),
+  beginMessage: z.string().min(1, "Begin message is required"),
+  numberToCall: z.string().optional(),
+  generalTools: z
+    .array(
+      z.object({
+        type: z.string(),
+        name: z.string(),
+        description: z.string().nullish().default(null),
+        cal_api_key: z.string().nullish().default(null),
+        event_type_id: z.number().nullish().default(null),
+        timezone: z.string().nullish().default(null),
+      })
+    )
+    .optional(),
+});
+
+const phoneNumberFormSchema = z.object({
+  phoneNumber: z.string().refine((val) => isValidPhoneNumber(val)),
+  terminationUri: z.string().min(1, "Termination URI is required"),
+  sipTrunkAuthUsername: z.string().optional(),
+  sipTrunkAuthPassword: z.string().optional(),
+  nickname: z.string().optional(),
+});
+
+type AgentFormValues = z.infer<typeof agentSchema>;
+type PhoneNumberFormValues = z.infer<typeof phoneNumberFormSchema>;
+// type RetellData = RouterOutputs["viewer"]["ai"]["get"]["retellData"];
+
+// type ToolDraft = {
+//   type: string;
+//   name: string;
+//   description: string | null;
+//   cal_api_key: string | null;
+//   event_type_id: number | null;
+//   timezone: string;
+// };
```

Comment:
We have decided that in the V1 we won't add the complexity of general tools. so i have commented this for now

Reply:
we would need all this in V2 so not deleting now